Source/Alamofire.swift

```diff
@@ -27,42 +27,83 @@ import Foundation
 /// Types adopting the `URLStringConvertible` protocol can be used to construct URL strings, which are then used to
 /// construct URL requests.
 public protocol URLStringConvertible {
-    /// A URL that conforms to RFC 2396.
+    /// A URL string that conforms to RFC 2396.
     ///
     /// Methods accepting a `URLStringConvertible` type parameter parse it according to RFCs 1738 and 1808.
     ///
     /// See https://tools.ietf.org/html/rfc2396
     /// See https://tools.ietf.org/html/rfc1738
     /// See https://tools.ietf.org/html/rfc1808
-    var urlString: String { get }
+    var urlString: String? { get }
+
+    /// Returns a URL string that conforms to RFC 2396 or throws if an `Error` was encoutered.
+    ///
+    /// Methods accepting a `URLStringConvertible` type parameter parse it according to RFCs 1738 and 1808.
+    ///
+    /// - throws: An `Error` if underlying url string is `nil`.
+    ///
+    /// - returns: A URL string.
+    func asURLString() throws -> String
+}
+
+extension URLStringConvertible {
+    public var urlString: String? { return try? asURLString() }
 }
 
 extension String: URLStringConvertible {
-    /// The URL string.
-    public var urlString: String { return self }
+    /// Returns a URL string that conforms to RFC 2396.
+    public func asURLString() throws -> String { return self }
```

Comment:
Seems like we want to do something here to ensure the string can be converted to a URL.